db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/PredicateList.java

```diff
@@ -1349,19 +1608,34 @@ void pullExpressions(int numTables,ValueNode searchClause) throws StandardExcept
             BooleanConstantNode trueNode=(BooleanConstantNode)getNodeFactory().getNode(C_NodeTypes.BOOLEAN_CONSTANT_NODE,
                     Boolean.TRUE,contextManager);
 
+            AndNode firstAndInProbeSet = null, prevAnd = null;
             while(topAnd.getRightOperand() instanceof AndNode){
                 /* Break out the next top AndNode */
                 thisAnd=topAnd;
-                topAnd=(AndNode)topAnd.getRightOperand();
-                thisAnd.setRightOperand(null);
-
-                /* Set the rightOperand to true */
-                thisAnd.setRightOperand(trueNode);
-
-                /* Add the top AndNode to the PredicateList */
-                newJBitSet=new JBitSet(numTables);
-                newPred=(Predicate)getNodeFactory().getNode(C_NodeTypes.PREDICATE,thisAnd,newJBitSet, contextManager);
-                addPredicate(newPred);
+                topAnd = (AndNode) topAnd.getRightOperand();
+                if (!thisAnd.isNextAndedPredInSameMultiprobeSet()) {
+                    /* Set the rightOperand to true */
+                    thisAnd.setRightOperand(trueNode);
+
+                    /* Add the top AndNode to the PredicateList */
+                    newJBitSet=new JBitSet(numTables);
+                    newPred=(Predicate)getNodeFactory().getNode(C_NodeTypes.PREDICATE,thisAnd,newJBitSet, contextManager);
+                    addPredicate(newPred);
```

Comment:
This logic does not look right, thisAnd node here could be part of the AndedPredInSameMultiprobeSet, but we are separating it out. This may explain why in the following explain we are seeing two duplicate inlists:
```
create table t1 (a1 int, b1 int, c1 int, d1 int, primary key(a1,b1));
splice> explain select * from t1 where a1=1 and b1=10 or a1=2 and b1=20;
Plan                                                                                                                                                                                                                                                            
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Cursor(n=4,rows=1,updateMode=READ_ONLY (1),engine=control)                                                                                                                                                                                                      
  ->  ScrollInsensitive(n=3,totalCost=8.046,outputRows=1,outputHeapSize=2 B,partitions=1)                                                                                                                                                                       
    ->  ProjectRestrict(n=2,totalCost=4.04,outputRows=1,outputHeapSize=2 B,partitions=1,preds=[((A1[0:1],B1[0:2]) IN ((1,10),(2,20))),((A1[0:1],B1[0:2]) IN ((1,10),(2,20)))])                                                                                  
      ->  TableScan[T1(1648)](n=1,totalCost=4.04,scannedRows=20,outputRows=20,outputHeapSize=2 B,partitions=1)                                                                                                                                                  

4 rows selected
```

Reply:
Yes, I've corrected the logic, thanks.